beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -317,21 +319,22 @@ func (f *blocksFetcher) handleRequest(ctx context.Context, start primitives.Slot
 	}
 
 	response.bwb, response.pid, response.err = f.fetchBlocksFromPeer(ctx, start, count, peers)
-	if response.err == nil {
-		if coreTime.PeerDASIsActive(start) {
-			bwb, err := f.fetchColumnsFromPeer(ctx, response.bwb, response.pid, peers)
-			if err != nil {
-				response.err = err
-			}
-			response.bwb = bwb
-		} else {
-			bwb, err := f.fetchBlobsFromPeer(ctx, response.bwb, response.pid, peers)
-			if err != nil {
-				response.err = err
-			}
-			response.bwb = bwb
-		}
+
+	if response.err != nil {
+		return response
+	}
+
+	if coreTime.PeerDASIsActive(start) {
+		response.err = f.fetchDataColumnsFromPeers(ctx, response.bwb, peers)
```

Comment:
can we follow the same pattern where we return the blocks with blobs / blocks with columns . So instead of returning just the error you return the blocks too

Reply:
Actually I changed it on purpose.

The reason was, this function (as well as `fetchBlobsFromPeer` ) mutates the `response.bwb` argument.

The fact that the function both changes the input `response.bwb` argument, and returns it is quite misleading for the function user. 

The user may think that the argument is not changed, and the mutated structure is returned.
(A little bit like `peers = filterPeers(peers, ...)` where in this case the peers in argument is not mutated.)


==> If we want to have consistency between functions, I would then advocate to change others functions not to return `response.bwb`.

Note: I added this in the function documentation:
```
// This function mutates `bwb` by adding the retrieved data columns.
```

Reply:
Fixed the opposite way in 41f2a8567b34a5d52374c4996bcb8adbba543d6a.